pyproject.toml

```diff
@@ -1,16 +1,14 @@
 [build-system]
 requires = [
-    "setuptools",
-    "scikit-build",
+    "scikit-build-core",
     "cython>=3",
-    "cmake",
-    "ninja",
     "numpy>=2.0.0",
 ]
-build-backend = "setuptools.build_meta"
+build-backend = "scikit_build_core.build"
 
 [project]
 name = "blosc2"
+version = "3.0.0-beta.2.dev0"
```

Comment:
`scikit-build-core` supports dynamic version. The [documented setup](https://scikit-build-core.readthedocs.io/en/latest/configuration.html#dynamic-metadata) should be sufficient for your use-case.